lib/holdingsiq.js

```diff
@@ -0,0 +1,345 @@
+      method: 'get',
+      url: `/${custid}/holdings/status`,
+      headers: {
+        'x-api-key': apikey,
+        'Content-Type': 'application/json',
+      },
+    });
+  } catch (err) {
+    logger.errorRequest(err);
+    process.exit(1);
+  }
+  return res?.data?.totalCount;
+};
+
+/**
+ * Parse the result of getHoldings for ezHLM format
+ * @param {Array<Object>} data Data to be parsed
+ * @param {String} institute Name of institute
+ * @param {String} index Index where the values will be inserted
+ * @returns {Object} Data ready to be inserted in elastic
+ */
+const parseGetHoldings = (data, institute, index) => {
+  const results = [];
+  data.forEach((e) => {
+    const result = {
+      ezhlmid: `${institute}-${e?.vendor_id}-${e?.package_id}-${e?.title_id}`,
+      PackageID: e?.package_id,
+      VendorID: e?.vendor_id,
+      KBID: e?.title_id,
+      PackageName: e?.package_name,
+      VendorName: e?.vendor_name,
+      Title: e?.publication_title,
+      AccessType: e?.access_type,
+      PublicationType: e?.publication_type,
+      ResourceType: e?.resource_type,
+      URL: e?.title_url,
+      Author: e?.first_author,
+      NumLastIssueOnline: e?.num_last_issue_online,
+      NumLastVolOnline: e?.num_last_vol_online,
+      CoverageDepth: e?.coverage_depth,
+      NumFirstIssueOnline: e?.num_first_issue_online,
+      NumFirstVolOnline: e?.num_first_vol_online,
+      DateFirstIssueOnline: e?.date_first_issue_online,
+      DateLastIssueOnline: e?.date_last_issue_online,
+      EmbargoInfo: e?.embargo_info,
+      OnlineIdentifier: e?.online_identifier,
+      PrintIdentifier: e?.print_identifier,
+    };
+    results.push({ update: { _index: index, _id: result.ezhlmid } });
+    results.push({ doc: result });
+  });
+  return results.slice();
+};
+
+/**
+ * getHoldings for enrich initialization
+ * @param {Object} conf - config on institute (name, apikey, custid)
+ * @param {*} count - number of documents to recover
+ * @param {*} offset - page
+ * @param {String} index Index where the values will be inserted
+ * @returns {Object} Data ready to be inserted in elastic
+ */
+const getHoldings = async (conf, count, offset, index) => {
+  const {
+    apikey,
+    custid,
+    name,
+  } = conf;
+
+  let res;
+
+  try {
+    res = await holdings({
+      method: 'get',
+      url: `/${custid}/holdings`,
+      params: { count, offset, format },
+      headers: {
+        'x-api-key': apikey,
+        'Content-Type': 'application/json',
+      },
+    });
+  } catch (err) {
+    logger.errorRequest(err);
+    process.exit(1);
+  }
+  const result = parseGetHoldings(res?.data?.holdings, name, index);
+  return result;
+};
+
+/**
+ * get the contributor according to the type
+ * @param {Array<Object>} contributors - contributors array
+ * @param {String} type - type of contributor
+ * @returns {String} name of contributor
+ */
+const getContributor = (contributors, type) => contributors.filter((contributor) => {
```

Comment:
`getContributor` is supposed to return a single element, but you return the result of `contributors.filter()`